README.md

```diff
@@ -257,3 +259,29 @@ Attestation workflow with differences from the proposed approach in **bold**:
 
+## Switch Owner Prod TLS Cert Issuance
+
+Although this is outside of the scope of this doc, the expectation is that each control card has its own separate prod TLS key pair and cert that it never shares with the other card.
+Each card can perform CSR-style TLS key pair/cert issuance, where TLS pair key is issued by a control card and the private key never leaves a given control card (never shares the key with another card within the same switch chassis either).
+Switch owner will always attest a given control card before issuing a new or rotating an existing prod TLS cert.
+In other words, **switch-owner-issued production TLS credentials/certs can only be accessible to control cards that have been TPM enrolled and attested by switch owner**. If a standby control card becomes active/primary, it must use its own TLS cert for all connections with switch owner infra.
+
+## RMA Scenario
+
+One benefit of having multiple control cards is a redundancy model where one control card (active) is serving traffic while another card is unavailable. In such a scenario a switch owner would typically want to replace the failed control card, while the active card is still serving traffic (aka hot-swapping).
+The newly inserted (standby) card must be TPM enrolled and attested by the switch owner before the card gets access to switch-owner-issued prod TLS cert. Thus, conceptually the RMA workflow would be the following:
+
+1. During the initial device secure install, switch owner TPM enrolls and attests both control cards. One of the cards acts as primary, serving prod traffic. Each control card has its own switch-owner-issued TLS cert.
+2. One of the cards fails and a new standby card is inserted, while the active card is serving prod traffic.
+3. Active control card conducts an auth handshake with a newly inserted control card.
+   1. Active card sends a nonce to the new standby card.
+   2. Standby card signs the nonce with its IDevID private key and sends it back along with its IDevID cert.
+   3. Active card verifies nonce signature and IDevID cert.
+4. Active card notifies switch owner infra that a new control card is inserted.
+5. Switch owner initiates enrollz and attestz workflow for the new standby card. Once attestz succeeds, switch owner issues to the standby control card its own TLS credentials/cert.
+
+Although this is highly undesirable, if per-control-card TLS cert is temporary unsupported and instead both control cards must share the same prod TLS key pair and cert, then the primary/active control card can only sync the TLS credentials/cert to the newly inserted standby/secondary card after the standby card have been enrolled and attested by the switch owner.
```

Comment:
the previous section seems to conflict with this one? either it is supported or it isn't?

Reply:
I don't think "if per-control-card TLS cert is temporary unsupported" suggests any contradictions with previous sections, but I am also ok with removing this paragraph altogether. If some of the vendors end up not being able to deliver "per-control-card TLS cert support" by the set timelines, we can deal with them on a case by case basis and consider suggesting a phased-out approach similar to the one offered in this paragraph. How does that sound?

Reply:
@morrowc  - Thoughts?

I am not clear this is actually the only model we should support from an openconfig perspective. This makes several assumptions about the only security models that are supported for OC.

I think we should make sure this more broadly encompasses all workflows that operators may want to utilize for securing thier devices.

We can suggest this is this preferred method of operation but i do think there are other scenarios which are still valid and actually more inline with current vendor hardware.

Significant amount of existing vendor hardware doesn't even have TPM's and do not have a mode which doesn't copy credentials between control cards.

Reply:
Thanks, Marcus. The thing is that we already discussed this, which is why [go/ncp-security-switch-requirements-2023](http://goto.google.com/ncp-security-switch-requirements-2023) has been finalized and  shared with all vendors. If you want to bring these concerns again, please raise them in that doc or directly with authors.

This README is only authoritative when it comes TPM enrollment and attestation, while Prod TLS cert issuance procedure is a separate workflow and is outside of the scope of this README (which is mentioned in the doc). If you want, I can highlight it in bold that this doc is **not** authoritative on TLS issuance workflow/APIs mechanics, but that it does mandate that "switch-owner-issued production TLS credentials/certs can only be accessible to control cards that have been TPM enrolled and attested by switch owner".

This paragraph ("Although this is highly undesirable, if per-control-card TLS cert is temporary unsupported...") is there to address cases where vendors cannot support "per-control-card TLS cert" requirement in the shorter term future. I am also ok with removing it completely as I suggested in the previous comment.

"Significant amount of existing vendor hardware doesn't even have TPM's". Since this repo is about TPM enrollment and attestation, it would not be relevant to such vendors.

Please let me know what you think and how the wording can be improved to better align with our broader agreement and long term goals.